source/extensions/filters/udp/dns_filter/dns_filter.cc

```diff
@@ -0,0 +1,55 @@
+#include "extensions/filters/udp/dns_filter/dns_filter.h"
+
+#include "envoy/network/listener.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace DnsFilter {
+
+DnsFilterEnvoyConfig::DnsFilterEnvoyConfig(
+    Server::Configuration::ListenerFactoryContext& context,
+    const envoy::config::filter::udp::dns_filter::v2alpha::DnsFilterConfig& config)
+    : root_scope(context.scope()), stats_(generateStats(config.stat_prefix(), root_scope)) {
+
+  using envoy::config::filter::udp::dns_filter::v2alpha::DnsFilterConfig;
+
+  // store configured data for server context
+  const auto& server_config = config.server_config();
+
+  if (server_config.has_control_plane_cfg()) {
+
+    const auto& cfg = server_config.control_plane_cfg();
+    const size_t entries = cfg.virtual_domains().size();
+
+    virtual_domains_.reserve(entries);
+    for (const auto virtual_domain : cfg.virtual_domains()) {
+      DnsAddressList addresses{};
+
+      if (virtual_domain.endpoint().has_addresslist()) {
+        const auto& address_list = virtual_domain.endpoint().addresslist().address();
+        addresses.reserve(address_list.size());
+        for (const auto& configured_address : address_list) {
+          addresses.push_back(configured_address);
+        }
+      }
+
```

Comment:
can we also add validation here that the list of virtual domain names are indeed part of the "known_domains"? That would be useful down the road when you come to the implementation of the filter query

Reply:
Let me add a TODO.   I will get the actual implementation in after I think about it for a bit.